lighthouse-core/lib/manifest-parser.js

```diff
@@ -242,6 +242,19 @@ function parseIcon(raw, manifestUrl) {
 
   const type = parseString(raw.type, true);
 
+  const purpose = {
```

Comment:
nit: feels a bit odd to construct the default this way with the default being in the `else` case, WDYT about...

Reply:
I originally had it the way you suggest below but I changed it realizing that I was ending up with duplicate 'any's if the purpose is defined with any as a value. In hindsight, this duplication doesn't pose much of an issue in this case.

Reply:
well there shouldn't be any duplication if we just `purpose.value = raw.split..` no need to do `.concat` :)

Reply:
I agree. Not sure why I hadn't considered just overwriting the value array in the case that purpose exists. Anyways, took your suggestion and changed a couple more things due to typing issues.